### PR TITLE
cholesky_solve for tensorflow frontend

### PR DESCRIPTION
**Pull request by agniiyer**

### CODE REVIEW

**Comment by agniiyer** [comment]
#3793

**Comment by Auxia** [comment]
Hi! Thanks for contributing to ivy. Please let me know when you are ready for a review :)

P.S. to fix merge conflicts you can run `merge_with_upstream .` script (at the top level of the ivy directory). It should fix it for you mostly. Also do not forget to install the [pre-commit](https://lets-unify.ai/ivy/contributing/0_setting_up.html#pre-commit) hook to help with lint errors.

**Comment by agniiyer** [comment]
Hi I'm ready for a review! I ran merge_with_upstream after installing pre-commit but my fork isn't updating when I push the changes. Maybe I'm missing something?

There are still some hypothesis errors being thrown though it doesn't say what the error is specifically. Any advice on how to fix it? Thanks! :)

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 1]
```diff
@@ -211,6 +211,39 @@ def test_tensorflow_slogdet(
...
+    ),
+)
+def test_tensorflow_cholesky_solve(
+    dtype_and_x,
+    as_variable,
+    num_positional_args,
+    native_array,
+    fw,
+):
+    data1, data2 = dtype_and_x
```
Comment (Auxia) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:229]: I see you have liberally tried to copy the test for the `solve` function. However that function itself isn't passing its tests. You might want to check out the Ivy Tests for the solve function in our ivy framework [here](https://github.com/unifyai/ivy/blob/c6e3c7a24f63fe7360a6977a3ca89b3ae24d9ad5/ivy_tests/test_ivy/test_functional/test_core/test_linalg.py#L621) to understand how to implement tests for function like these. Along with the [docs](https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends_tests.html#ivy-frontend-tests)

**Review by Auxia — CHANGES_REQUESTED** [review]
As I have mentioned in the review comments. You will need to rethink the way you are implementing this function as Ivy framework itself does not currently have a cholesky_function. But it is possible to create a composition of other ivy functions to implement it.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 1 above)
Comment (agniiyer) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:229]: @Auxia Thanks! Can you explain the difference between the functional and frontend tests?

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 2]
```diff
@@ -34,6 +34,13 @@ def slogdet(input, name=None):
 slogdet.unsupported_dtypes = ("float16", "bfloat16")
 
 
+def cholesky_solve(chol, rhs, name=None):
+    return ivy.solve(chol, rhs)
```
[the remark this answers is not in the document]
Comment (agniiyer) [reply ivy/functional/frontends/tensorflow/linalg.py]: @Auxia Hey I've updated this function! I am using solve again but it is now more appropriate for the cholesky approach.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 1 above)
Comment (agniiyer) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: @Auxia Hi thanks for the link to the ivy test for solve! I'm trying to use the same approach here. As I understand it, rhs can be the same as get_second_matrix() but I need to modify get_first_matrix() to return a symmetric positive definite matrix. I'm facing some Assertion Errors with the function right now and was hoping you can help me debug it? Thanks!

Comment (agniiyer) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py]: Thanks for the advice! I've updated the test to produce the cholesky decomposition of a symmetric positive definite matrix for testing. All tests are passing now!

**Pull request merged**